src/Parallel/shortestpaths/dijkstra.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    struct Parallel.MultipleDijkstraState{T, U}
+
+An [`AbstractPathState`](@ref) designed for Parallel.dijkstra_shortest_paths calculation.
+"""
+struct MultipleDijkstraState{T <: Real,U <: Integer} <: AbstractPathState
+    dists::Matrix{T}
+    parents::Matrix{U}
+end
+
+"""
+    Parallel.dijkstra_shortest_paths(g, sources=vertices(g), distmx=weights(g))
+
+Compute the shortest paths between all pairs of vertices in graph `g` by running
+[`dijkstra_shortest_paths`] for every vertex and using an optional list of source vertex `sources` and
+an optional distance matrix `distmx`. Return a [`Parallel.MultipleDijkstraState`](@ref) with relevant
+traversal information.
+"""
+function dijkstra_shortest_paths(g::AbstractGraph{U},
```

Comment:
This would work as `Parallel.dijkstra_shortest_path(g, [1])` and not as `Parallel.dijkstra_shortest_paths(g, 1)`. So we should add something like
```
dijkstra_shortest_paths(g::AbstractGraph, src::Integer, distmx::AbstractMatrix=weights(g); allpaths=false, trackvertices=false) =
dijkstra_shortest_paths(g, [src;], distmx; allpaths=allpaths, trackvertices=trackvertices)
```

Reply:
Yes. But won't this create confusions wrt to arguments of sequential Dijkstra? This was one of the reasons I named this function `multisource_dijkstra_shortest_path`.

Reply:
It might but I think that a methoderror is preferred here. This algorithm is much slower than single source dijkstra for a single source and it’s invariably a mistake to call it with one vertex. 

When we start dispatching shortest paths based on an algorithm type (think `shortest_path(g, 1; alg=Dijkstra()`), we might need to revisit. But right now I think calling Parallel.foo() with arguments that don’t parallelize should be an error.

Reply:
The only problem with @sbromberger's approach is that we can't measure the overhead incurred by Parallel.foo. I am thinking of a paper where I run sequential and parallel algorithms and measure the performance improvement.

Actually, it is a scientific mistake to call a parallel algorithm with one thread and use that as the serial runtime, you must run the optimal sequential algorithm. Maybe we can help enforce that methodology here by rejecting arguments that lead to parallel algorithms being used in a way that we know won't yield speedup?

It is better to raise an error that tells the user "Parallel.foo doesn't support a single vertex source. Use the sequential LightGraphs.foo instead, or file a PR with a new parallel implementation that accelerates the single source case"

Reply:
@sbromberger raising error would be better

Reply:
I'd normally agree but I think raising a custom error conflicts with #789. In the interest of getting this into 1.0, let's leave it as-is, but we can revisit afterwards since it will be a non-breaking change.

Reply:
sure!